src/main/java/stream/flarebot/flarebot/util/pagination/PagedEmbedBuilder.java

```diff
@@ -0,0 +1,124 @@
+package stream.flarebot.flarebot.util.pagination;
+
+import net.dv8tion.jda.core.EmbedBuilder;
+import net.dv8tion.jda.core.entities.MessageEmbed;
+
+public class PagedEmbedBuilder<T> {
+    private String title;
+    private String codeBlock;
+    private PaginationList<T> list;
+    private boolean hasCodeBlock = false;
+
+    /**
+     * Instantiates the builder with a {@link PaginationList}.
+     *
+     * @param list The {@link PaginationList} to use as the pages.
+     */
+    public PagedEmbedBuilder(PaginationList<T> list) {
+        this.list = list;
+    }
+
+    /**
+     * Sets the title for the Embed.
+     *
+     * @param title The title.
+     * @return this.
+     */
+    public PagedEmbedBuilder setTitle(String title) {
+        this.title = title;
+        return this;
+    }
+
+    /**
+     * Sets the type of code block to use.
+     *
+     * @param codeBlock The string representing the code block.
+     * @return this
+     */
+    public PagedEmbedBuilder setCodeBlock(String codeBlock) {
+        this.codeBlock = codeBlock;
+        return this;
+    }
+
+    /**
+     * Enables code blocks
+     * If you use {@link PagedEmbedBuilder#setCodeBlock(String)} it auto enables.
+     *
+     * @return this
+     */
+    public PagedEmbedBuilder enableCodeBlock() {
+        this.hasCodeBlock = true;
+        return this;
+    }
+
+    /**
+     * Builds a {@link PagedEmbed} for use in embed pages.
+     *
+     * @return {@link PagedEmbed}.
+     */
+    public PagedEmbed build() {
+        boolean pageCounts = false;
+        if (list.getPages() > 1) {
+            pageCounts = true;
+        }
+        if (codeBlock != null) {
+            hasCodeBlock = true;
```

Comment:
Any reason why you do this here rather than in setCodeBlock?

Reply:
Lazy

Reply:
Lazy? This is more code.

Reply:
I mean when it's old code yes it is being lazy cause i didn't delete it when i switched stuff. :P